clang/include/clang/Basic/DiagnosticLexKinds.td

```diff
@@ -932,6 +932,12 @@ def warn_module_conflict : Warning<
   InGroup<ModuleConflict>;
 
 // C++20 modules
+def err_module_decl_cannot_be_macros : Error<
+  "the name of a module%select{| partition}0 declaration cannot contains "
+  "an object-like macro %1, and the macro will not expand"
+  "%select{|; did you mean '%3'?}2">;
```

Comment:
Not sure if you need "and the macro will not expand" here. Per the spec you don't even get to if expansion happens or not, it's just invalid being a macro.

```suggestion
def err_module_decl_cannot_be_macros : Error<
  "the module name in a module%select{| partition}0 declaration cannot contain "
  "an object-like macro %1%select{|; did you mean '%3'?}2">;
```

Reply:
Agree, with a little improve ` it's consists of one or more identifiers separated by '.'`.